drivers/i2c/i2c_nrfx_twi.c

```diff
@@ -148,6 +150,40 @@ static int init_twi(struct device *dev, const nrfx_twi_config_t *config)
 
 	return 0;
 }
+
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+static int twi_nrfx_pm_control(struct device *dev, u32_t ctrl_command,
+				void *context, device_pm_cb cb, void *arg)
+{
+	static u32_t current_state = DEVICE_PM_ACTIVE_STATE;
+
+	if (ctrl_command == DEVICE_PM_SET_POWER_STATE) {
+		u32_t new_state = *((const u32_t *)context);
+
+		if (new_state != current_state) {
+			if (new_state == DEVICE_PM_ACTIVE_STATE) {
```

Comment:
Could we have switch/case here? Please note that "if" requires priority encoding which is not needed for switch case so the resulting code is smaller and faster.